### ISSUE TITLE
Detect render() directly in describe block body

### ISSUE THREAD

**Issue by robatwilliams** [testing-library/eslint-plugin-testing-library#345]
I'm helping a team adopt React Testing Library in a codebase that already uses Enzyme.

We have quite a lot of the following kind of pattern (with multiple levels of nesting), which is bad and has all kinds of issues, but works in Enzyme:
```javascript
describe('x', () => {
  const wrapper = mount(<Component />);

  it('y', () => {});

  it('z', () => {});
});
```

In React Testing Library, due to the shared global nature of document/screen, doing that will cause leakage between tests.

Maybe `no-render-in-setup` could be extended to cover this?

I don't think any code directly in the describe body is good, so maybe a rule in eslint-plugin-jest for that would be better and solve the more general problem.

**Comment by Belco90** [comment]
Hey @robatwilliams, thanks for your feedback!

This sounds like a good idea indeed. There are few things to keep in mind tho:
- We can't opt-in this new error reported by default in `no-render-in-setup` since that would mean a breaking change. In order to keep it as a non-breaking change release, we need to include this as "A new option to an existing rule that does not result in ESLint reporting more linting errors by default."
- In order to achieve the previous point, we need to be able to google this new error with a new option (disabled by default)
- Current `"allowTestingFrameworkSetupHook"` doesn't feel the proper option to config this new `describe` reporting, so we would need to add a new option for it. Something like `"reportDescribeBlock": "on" | "off"`? (default to "off")

What do you think? And more important: would you like to give it a try?

**Comment by robatwilliams** [comment]
Makes sense.

I'm not sure this would be hugely valuable unless someone is converting some old tests, or isn't familiar with the basics of test isolation. So it's unlikely I'll work on this. My current solution has been to find all these cases with a regex and add a comment to them.

**Comment by Belco90** [comment]
Fair enough! I'd say for the same reasons this is a low priority for us, but definitely something we can work on when other pending stuff gets done.

**Comment by robatwilliams** [comment]
I'm going to close this myself. It's such a basic issue that anybody writing tests should understand.
